src/test_pages/BasePage.ts

```diff
@@ -12,7 +12,7 @@ export abstract class BasePage {
   protected async clickOnElement(element: Locator, timeout?: number) {
     // TODO: make sure that element.toString() doesn't make [object Object]
     await test.step(`Click on ${element.toString()}`, async () => {
-      await element.click({ timeout: timeout || 0 })
+      await element.click({ timeout })
```

Comment:
```suggestion
      expect(element).not.toBeNull()
      await element.click({ timeout })
```
I think the problem is that element.click() bypasses the ```expect``` timeout of 5 seconds, leaving it to hang until the ```test``` timeout

Reply:
I'm pretty sure that `actionTimeout` is `5000ms` by default.
`element` is a Locator object, so it won't be null even if the locator won't resolve to any dom element. You probably mean you want to do something like `expect(locator).toBeVisible()` or `expect(locator).toBeEnabled()`, but the [auto-waiting](https://playwright.dev/docs/actionability) magic will do it for us so it's not necessary